appcues/src/main/java/com/appcues/analytics/AnalyticsTracker.kt

```diff
@@ -96,8 +96,6 @@ internal class AnalyticsTracker(
     }
 
     private fun updateAnalyticsFlow(type: AnalyticType, isInternal: Boolean, activity: ActivityRequest) {
-        appcuesCoroutineScope.launch {
-            _analyticsFlow.emit(TrackingData(type, isInternal, activity))
-        }
+        _analyticsFlow.tryEmit(TrackingData(type, isInternal, activity))
```

Comment:
updated this flow to have a max size buffer (L22 above) and now call `tryEmit` here, which is non-suspending.

what I ran into, without this, is that analytics can get sent into the flow out of order, when using the [`UnconfinedTestDispatcher` ](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/-unconfined-test-dispatcher.html)

this is because we have multiple spots that can trigger calls into this class during flow execution - step interaction actions and experience lifecycle observer.

it _seems_ safe here to just use the non-suspending version with a large buffer, since we'll never get anywhere near that buffer size in practice, and the consumers of this flow are quite simple - our own debugger, and the analytics publisher that sends to any attached listeners.